### ISSUE TITLE
finalize command crashes with message "double free or corruption (out)"

### ISSUE THREAD

**Issue by lneely** [lneely/pcloudcc-lneely#21]
# environment

**OS:** Linux (Artix)
**Compiler:** gcc (GCC) 14.1.1 20240522
**sqlite:** 3.46.0-1
**zlib:** 1:1.3.1-2
**mbedtls:** 1.3.10

**Special Considerations (e.g., using branch other than main, running in a VM/container):** 
N/A

# repro steps

In terminal
```
pcloudcc -u me@myemail.com`
```

In another terminal
```
pcloudcc -k
> finalize
```

# expected behavior

`pcloudcc` daemon exits and command prompt terminates normally.

# actual behavior

`pcloudcc` daemon crashes, command prompt displays error output:

```
Finalize failed. return is 32767 and message is P
double free or corruption (out)
Aborted
```

**Comment by lneely** [comment]
As I feared, I'm getting different error messages under different conditions. Sometimes it's invalid size, sometimes invalid pointer, and sometimes double free or corruption. I suspect a race condition of some kind, perhaps one thing is freeing the pointer and another thing is trying to free it again. This is going to be a pain in the ass... :sweat_smile: 

Unfortunately, this means I can't accept this pull request just yet. The underlying problem needs to be fixed first. If you're willing to dig in, go for it.

**Comment by lneely** [comment]
This issue is fixed with PR #29.
